test/sqllogictest/list.slt

```diff
@@ -2575,3 +2575,262 @@ SELECT LIST[LIST[customer.first_name, customer.last_name], LIST[customer.zip], L
 ----
 {{alice,lasta},{10003},{1}}
 {{charlie,lastc},{11217},{3}}
+
+## Optimization: Reducing ListIndex(ListCreate, literal) and multidimensional variations
+# Most of these tests could also be in src/expr/tests/testdata/reduce, but writing these big ListCreate expressions is a
+# bit cumbersome in json.
+
+statement ok
+CREATE TABLE t3(f1 int, f2 int, f3 int, f4 int, f5 int, f6 int, f7 int, f8 int, n int, m int, l int list)
+
+statement ok
+INSERT INTO t3 VALUES (1, 2, 3, 4, 5, 6, 7, 8, 1, 2, list[42, 43])
```

Comment:
And/Or you could materialize the table so that the plan doesn't contain the extra
```
Source materialize.public.t3 (u27):
| Project (#5)

Query:
```
lines, which would make the output easier to interpret.

Reply:
Both done.

But for a few of the tests, I didn't add the output checking:
- For the "index is out of bounds" cases I check the output for only one of them.
- The last test has a non-deterministic output, because of the `limit 1` without sorting. (If I added `order by` then the plan changed, and the optimization was no longer applying.)